packages/client/hmi-client/src/components/workflow/tera-calibration-node-ciemss.vue

```diff
@@ -288,13 +264,30 @@ const getStatus = async (simulationId: string) => {
 	}
 	console.log(`Simulation Id:${simulationId}`);
 	const runIds = [simulationId];
+
+	// open a connection for each run id and handle the messages
+	runIds.forEach((id) => {
+		eventSourceManager.openConnection(id, `/api/simulations/${id}/partial-result`);
+		eventSourceManager.setMessageHandler(id, (message) => {
+			const parsedMessage = JSON.parse(message);
+			if (parsedMessage.progress) {
+				progress.value.value = Math.round(parsedMessage.progress * 100);
+			}
+		});
```

Comment:
The function would be clearer if it was clearer defined with a named and then passed as a parameter. In my opinion.
```js
function handlingProgress(message) {...}
[...]
eventSourceManager.setMessageHandler(id, handlingProgress);
```